pkg/scaffold/v2/internal/string_utils.go

```diff
@@ -33,44 +33,40 @@ import (
 // v1 will be inserted below the lines containing m1 string and v2 will be inserted
 // below line containing m2 string.
 func insertStrings(r io.Reader, markerAndValues ...string) (io.Reader, error) {
-	if len(markerAndValues)%2 != 0 {
-		return nil, fmt.Errorf("invalid marker and value pairs")
-	}
+	// reader clone is needed since we will be reading twice from the given reader
+	buf := new(bytes.Buffer)
+	rClone := io.TeeReader(r, buf)
 
-	mvPairs := map[string]string{}
-	for i, s := range markerAndValues {
-		if i%2 == 0 {
-			mvPairs[s] = markerAndValues[i+1]
-		}
+	markerVals, err := filterExistingValues(rClone, markerAndValues...)
+	if err != nil {
+		return nil, err
 	}
 
-	buf := new(bytes.Buffer)
+	out := new(bytes.Buffer)
 
-	scanner := bufio.NewScanner(r)
+	scanner := bufio.NewScanner(buf)
 	for scanner.Scan() {
 		line := scanner.Text()
 
-		for m, v := range mvPairs {
-			if strings.TrimSpace(line) == strings.TrimSpace(v) {
-				// since value already exist, so avoid duplication
-				delete(mvPairs, m)
-			}
-			if strings.Contains(line, m) {
-				_, err := buf.WriteString(v)
-				if err != nil {
-					return nil, err
+		for marker, vals := range markerVals {
+			if strings.Contains(line, marker) {
```

Comment:
I'm not sure `strings.Contains` is correct here -- that means that `// +foo:marker:bar` would match `foo:marker`, which seems incorrect.

Reply:
Good catch. Added strict comparison.